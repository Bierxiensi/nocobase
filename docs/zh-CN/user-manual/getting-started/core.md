# 核心概要

NocoBase 的核心功能分别对应右上角的三个图标，分别为：

- 界面设计
- 插件管理器
- 配置中心

```bash
[图：右上角图标截图]
```

## 界面设计

界面设计包括两部分

1. 菜单
2. 页面

```bash
[图：菜单和页面]
```

## 插件管理器

分为本地、内置和插件市场，本地插件可以激活、禁用和删除，内置插件为系统的核心组成部分不可禁用；插件市场敬请期待。

```bash
[图：插件管理器]
```

## 配置中心

通过插件为平台提供系统级别的配置项，如数据表配置、权限配置、系统设置等。

```bash
[图：配置中心]
```