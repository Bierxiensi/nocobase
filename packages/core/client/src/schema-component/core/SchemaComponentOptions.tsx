import { ExpressionScope, SchemaComponentsContext, SchemaOptionsContext } from '@formily/react';
import React, { useContext } from 'react';
import { useApp } from '../../application-v2';
import { ISchemaComponentOptionsProps } from '../types';

export const useSchemaOptionsContext = () => {
  const options = useContext(SchemaOptionsContext);
  return options || {};
};

export const SchemaComponentOptions: React.FC<ISchemaComponentOptionsProps> = (props) => {
  const { children } = props;
  const options = useSchemaOptionsContext();
  const app = useApp();
  const components = { ...app.components, ...options.components, ...props.components };
  const scope = { ...app.scopes, ...options.scope, ...props.scope };
  return (
    <SchemaOptionsContext.Provider value={{ scope, components }}>
      <SchemaComponentsContext.Provider value={components}>
        <ExpressionScope value={scope}>{children}</ExpressionScope>
      </SchemaComponentsContext.Provider>
    </SchemaOptionsContext.Provider>
  );
};
