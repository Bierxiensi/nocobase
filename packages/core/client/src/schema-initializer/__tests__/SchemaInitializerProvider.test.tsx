import React, { useContext } from 'react';
import { render } from '@testing-library/react';
import { connect } from '@formily/react';

import { SchemaComponentProvider } from '../../schema-component/core/SchemaComponentProvider';
import { SchemaComponent } from '../../schema-component/core/SchemaComponent';
import { SchemaInitializerProvider, SchemaInitializerContext } from '../SchemaInitializerProvider';

function App() {
  const ComponentA = connect(() => {
    const initializes = useContext(SchemaInitializerContext);
    return (
      <div>
        {Object.keys(initializes).map((v, index) => {
          return v;
        })}
      </div>
    );
  });
  const schema = {
    type: 'object',
    properties: {
      componentA: {
        type: 'void',
        'x-decorator': 'SchemaInitializerProvider',
        'x-component': 'ComponentA',
      },
    },
  };
  return (
    <SchemaComponentProvider components={{ ComponentA, SchemaInitializerProvider }}>
      <SchemaComponent schema={schema} />
    </SchemaComponentProvider>
  );
}
describe('SchemaInitializerProvider', () => {
  it('SchemaInitializer', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
