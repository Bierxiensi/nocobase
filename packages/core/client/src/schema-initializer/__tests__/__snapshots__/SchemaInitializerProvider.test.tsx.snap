// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaInitializerProvider SchemaInitializer 1`] = `
<div>
  <div>
    BlockInitializers
    BulkEditFormItemInitializers
    CalendarActionInitializers
    CreateFormBlockInitializers
    CreateFormBulkEditBlockInitializers
    CustomFormItemInitializers
    DetailsActionInitializers
    FormActionInitializers
    CreateFormActionInitializers
    UpdateFormActionInitializers
    BulkEditFormActionInitializers
    FormItemInitializers
    KanbanActionInitializers
    ReadPrettyFormActionInitializers
    CalendarFormActionInitializers
    ReadPrettyFormItemInitializers
    RecordBlockInitializers
    RecordFormBlockInitializers
    SubTableActionInitializers
    TableActionColumnInitializers
    TableActionInitializers
    TableColumnInitializers
    TableSelectorInitializers
    TabPaneInitializers
    TabPaneInitializersForCreateFormBlock
    TabPaneInitializersForBulkEditFormBlock
  </div>
</div>
`;
